### PR TITLE
Adaptive Sync support

### PR DESCRIPTION
**Pull request by IMS212**
This utilizes and checks for the extensions GLX_EXT_swap_control_tear or WGL_EXT_swap_control_tear and sets the swap interval negative if so, allowing for vertical blanking to happen mid-frame/faster (or in some cases not happen at all) and reducing latency.

### CODE REVIEW

**Comment by burgerindividual** [comment]
Just to make sure, were there any issues reported with this pr that would stop me from merging this?

**Comment by IMS212** [comment]
As far as I'm aware, no.

**Comment by FlashyReese** [comment]
You need to call the V-Sync setter again because the window hint doesn't get updated until `setVsync` is called.

File: src/main/java/me/jellysquid/mods/sodium/client/gui/SodiumGameOptionPages.java [hunk 1]
```diff
@@ -306,6 +307,15 @@ public static OptionPage performance() {
                         .setFlags(OptionFlag.REQUIRES_RENDERER_UPDATE)
                         .build()
                 )
+                .add(OptionImpl.createBuilder(boolean.class, sodiumOpts)
+                        .setName(new TranslatableText("sodium.options.use_adaptive_sync.name"))
+                        .setTooltip(new TranslatableText("sodium.options.use_adaptive_sync.tooltip"))
+                        .setControl(TickBoxControl::new)
+                        .setImpact(OptionImpact.VARIES)
+                        .setEnabled(GLFW.glfwExtensionSupported("GLX_EXT_swap_control_tear") || GLFW.glfwExtensionSupported("WGL_EXT_swap_control_tear"))
+                        .setBinding((opts, value) -> opts.performance.useAdaptiveSync = value, opts -> opts.performance.useAdaptiveSync)
```
Comment (FlashyReese) [inline src/main/java/me/jellysquid/mods/sodium/client/gui/SodiumGameOptionPages.java]:
```suggestion
                        .setBinding((opts, value) -> {
                            opts.performance.useAdaptiveSync = value;
                            // Update the swap interval
                            MinecraftClient.getInstance().getWindow().setVsync(MinecraftClient.getInstance().options.getEnableVsync().getValue());
                        }, opts -> opts.performance.useAdaptiveSync)
```

**Review by FlashyReese — APPROVED** [review]

**Pull request merged**